File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        size_t n=nums.size(),l=0,r=0;

        while(r<n){
            nums[l] = nums[r];
            while(r<n && nums[l]==nums[r]){
                r++;
            }
            l++;
        }

        return l;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        

        frequency = {}
        #loop thru the array
        #map the number to their frequency
        for num in nums:
            if num not in frequency:
                frequency[num] = 1
            else:
                frequency[num] += 1


        sorted_keys = [key for key, value in sorted(frequency.items(), key=lambda item: item[1], reverse=True)]


        return list(sorted_keys)[:k]

        
        